cpp/src_prims/distance/euclidean.h

```diff
@@ -150,9 +150,8 @@ void euclideanAlgo2(Index_ m, Index_ n, Index_ k, const InType *pA,
                GemmEpilogueTraits_, GemmEpilogue_>(
     transa, transb, gemm_m, gemm_n, k, (EffOutType)1, aPtr, lda, bPtr, ldb,
     (EffOutType)0, nullptr, ldd, pDCast,
-    [enable_sqrt] HD(EpiParams & p) {
-      int err = p.initializeExtra(nullptr, nullptr, enable_sqrt);
-      return err;
+    [enable_sqrt] (typename EpilogueFunctor_::Params & p) {
```

Comment:
We also have similar such thing being passed in `algo1.h` and `l1.h` header as well. Why is that not causing a seg-fault?

Reply:
Oh yep let me check again today / now :)
It might be just the `__host__ __device__` issue (ie change `__host__ __device__` to `__host__` only), and the `typename` was just an extra thing I did.
I'll verify if removing it will help.

Reply:
@teju85  Oh it seems like only removing `__host__ __device__` (ie HD) solves the issue.

Reply:
So the typename thing I did was just tests, and I probably rushed to a conclusion...

Reply:
I'm currently trying to narrow down to a test case, I'll report back